src/test/ui/async-await/suggest-missing-await.rs

```diff
@@ -54,4 +54,21 @@ async fn suggest_await_on_match_expr() {
     };
 }
 
+async fn dummy_result() -> Result<(), ()> {
+    Ok(())
+}
+
+#[allow(unused)]
+async fn suggest_await_in_generic_pattern() {
+    match dummy_result() {
+        //~^ HELP consider `await`ing on the `Future`
+        //~| HELP consider `await`ing on the `Future`
+        //~| SUGGESTION .await
+        Ok(_) => {}
+        //~^ ERROR mismatched types [E0308]
+        Err(_) => {}
+        //~^ ERROR mismatched types [E0308]
+    }
+}
+
```

Comment:
I wonder, can this test be marked as `// run-rustfix`?

Reply:
`// run-rustfix` seems to produce broken output. 

1. In `suggest_await_in_async_fn_return`, the "add a semicolon" and "add .await" suggestions get applied out of order to produce `dummy();.await`
2. In `suggest_await_in_generic_pattern` (the test I added), it adds `.await` twice, because there are two type errors due to the `Ok` and `Err` branches.

Reply:
Ah, I see. Fair enough, leave as is.